File: metro-front-main/src/services/seller.ts
import api from "./api";

export class SellerService {
    static async createSeller(data) {
        return api.post("/sellers", {
            name: data.name,
            document: data.document,
            email: data.email,
            emailOther: data.emailSecondary,
            phone: data.phone,
            otherPhone: data.otherPhone,
            bank: data.bank ? data.bank : "",
            accountNumberBank: data.accountNumberBank ? data.accountNumberBank : "",
            accountBank: data.accountBank ? data.accountBank : "",
            creci: data.creci,
            pix: data.pix,
        });
    }

    static async updateSeller(data, id) {
        return api.put(`/sellers/${id}`, {
            name: data.name,
            document: data.document,
            email: data.email,
            emailOther: data.emailSecondary,
            phone: data.phone,
            otherPhone: data.otherPhone,
            bank: data.bank,
            accountBank: data.accountBank,
            accountNumberBank: data.accountNumberBank,
            creci: data.creci,
            pix: data.pix,
        });
    }


    static async getSellerByName(seller: any) {
        let name = seller.name.toUpperCase()
        return api.get(`/sellers?name=${name}`);
    }


    static async deleteClient(id: any) {
        return api.delete(`/sellers/${id}`)
    }


    static async getSeller(id) {
        return api.get(`/sellers/${id}`);
    }

    static async getSellers() {
        return api.get("/sellers");
    }
}

File: metro-api-main/src/main/kotlin/br/com/horys/metro/models/Seller.kt
package br.com.horys.metro.models

import br.com.horys.metro.models.validator.CpfCnpj
import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank

@Entity
@Table(name = "sellers")
data class Seller(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,
    val name: String,
    val externalId: String,
    val creci: String?,
    @field:CpfCnpj val document: String,
    @field:Email val email: String,
    @field:Email val emailOther: String?,
    @field:NotBlank val phone: String,
    val otherPhone: String?,
    val pix: String?,
    val bank: String?,
    val accountBank: String?,
    val accountNumberBank: String?,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now(),
    val active: Boolean = true,
    val deletedAt: LocalDateTime? = null
)